### DISCUSSION TITLE
how to start reading the code?

### DISCUSSION THREAD

**Discussion by zhumanklv** [npm/feedback#812] [General]
Hey npm team,

I want to start contributing to the npm codebase. However, I do not have prior experience of creating terminal commands or anything related to this. 

Where can I get the info regarding the architectural design of the codebase(i guess that's also needed)?

What do you suggest first to be able to understand the npm codebase quicker?

**Comment by Preetiraj3697** [comment]
Go to this. This is very useful for begginer
https://www.freecodecamp.org/news/npm-cheat-sheet-most-common-commands-and-nvm/
